libs/remix-ui/editor/src/lib/providers/definitionProvider.ts

```diff
@@ -30,20 +39,33 @@ export class RemixDefinitionProvider implements monaco.languages.DefinitionProvi
     */
     async jumpToPosition(position: any) {
         const jumpToLine = async (fileName: string, lineColumn: any) => {
+            const fileTarget = await this.props.plugin.call('fileManager', 'getPathFromUrl', fileName)
             if (fileName !== await this.props.plugin.call('fileManager', 'file')) {
                 await this.props.plugin.call('contentImport', 'resolveAndSave', fileName, null)
-                await this.props.plugin.call('fileManager', 'open', fileName)
+                const fileContent = await this.props.plugin.call('fileManager', 'readFile', fileName)
+                try {
+                    await this.props.plugin.call('editor', 'addModel', fileTarget.file, fileContent)
```

Comment:
you do that for making sure monaco editor can actually display the file?
could you add a comment for describing what this is about?

Reply:
it's for peeking. 
we capture the open file action from the editor but that doesn't happen when it's being peeked at. so we need to model, just like I do in the referenceProvider.